bin/propolis-server/src/lib/spec/builder.rs

```diff
@@ -25,12 +30,27 @@ use propolis_api_types::instance_spec::{
     v0::NetworkBackendV0,
 };
 
-use super::{ParsedNetworkDevice, ParsedStorageDevice};
+use crate::config;
+
+use super::{
+    api_request::{self, DeviceRequestError},
+    config_toml::{ConfigTomlError, ParsedConfig},
+    ParsedNetworkDevice, ParsedStorageDevice,
+};
+
+#[cfg(feature = "falcon")]
+use super::ParsedSoftNpu;
 
 /// Errors that can arise while building an instance spec from component parts.
 #[allow(clippy::enum_variant_names)]
 #[derive(Debug, Error)]
 pub(crate) enum SpecBuilderError {
+    #[error("error parsing config TOML")]
+    ConfigToml(#[from] ConfigTomlError),
+
+    #[error("error parsing device in ensure request")]
```

Comment:
should these include the underlying error?
```suggestion
    #[error("error parsing config TOML: {0}")]
    ConfigToml(#[from] ConfigTomlError),

    #[error("error parsing device in ensure request: {0}")]
```

or do they always get formatted using the verbose format that will display the whole source chain?

Reply:
There's only one caller who actually consumes this error, so I switched it to use the `#?` specifier.